### PR TITLE
Add parallel, serial, and slow GKE Jenkins jobs.

### PR DESCRIPTION
**Pull request by spxtr**
ref #22672

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -259,6 +259,39 @@
...
+        export JENKINS_PUBLISHED_VERSION="ci/latest-1.2"
+    suffix:
+        - 'gke-release-1.2':
+            description: 'Run E2E tests on GKE from the release-1.2 branch.'
+            timeout: 30
+            job-env: |
+                export PROJECT="k8s-jkns-e2e-gke-1-2"
+                export GINKGO_TEST_ARGS="--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]"
+                export GINKGO_PARALLEL="y"
+        - 'gke-serial-1.2':
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: did you intentionally drop "release" from this job and the next?

Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: I did, but on second thought it's better to keep it consistent with the gce jobs.

**Comment by spxtr** [comment]
@k8s-bot e2e test this issue: #21484

**Comment by spxtr** [comment]
@ixdy ptal

**Comment by ixdy** [comment]
LGTM

**Comment by spxtr** [comment]
Manual merge since I want to see if these work.

**Pull request merged**